source/StartConditionsPanel.cpp

```diff
@@ -0,0 +1,275 @@
+/* StartConditionsPanel.cpp
+Copyright (c) 2020 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "StartConditionsPanel.h"
+
+#include "Color.h"
+#include "Command.h"
+#include "ConversationPanel.h"
+#include "DataFile.h"
+#include "Dialog.h"
+#include "Files.h"
+#include "FillShader.h"
+#include "Font.h"
+#include "FontSet.h"
+#include "Format.h"
+#include "GameData.h"
+#include "GameWindow.h"
+#include "Information.h"
+#include "Interface.h"
+#include "LoadPanel.h"
+#include "MainPanel.h"
+#include "Messages.h"
+#include "PlayerInfo.h"
+#include "Planet.h"
+#include "Preferences.h"
+#include "Rectangle.h"
+#include "Ship.h"
+#include "ShipyardPanel.h"
+#include "StarField.h"
+#include "System.h"
+#include "UI.h"
+#include "WrappedText.h"
+
+#include <SDL2/SDL.h>
+
+#include "gl_header.h"
+
+using namespace std;
+
+StartConditionsPanel::StartConditionsPanel(PlayerInfo &player, UI &gamePanels, LoadPanel *loadPanel)
+	: player(player), gamePanels(gamePanels), loadPanel(loadPanel)
+{
+	if(GameData::Start().size())
+	{
+		chosenStart = GameData::Start()[GameData::Start().size()-1];	
+		hasChosenStart = true;
+	}
+	
+	descriptionBox = GameData::Interfaces().Get("start conditions menu")->GetBox("start description");
```

Comment:
GetBox returns a (0, 0, 0, 0) box when it's not found, so it doesn't matter (since it will make the box not display in the panel, which is what we want).